packages/dds/map/src/test/mocha/directory.spec.ts

```diff
@@ -1475,14 +1475,15 @@ describe("Directory", () => {
                 assert.ok(expectedDirectories2.size === 0);
             });
 
-            it("Only creates a subdirectory once", () => {
+            it("Only creates a subdirectory with same name once", () => {
                 const fooDirectory = directory1.createSubDirectory("foo");
                 fooDirectory.set("testKey", "testValue");
-                const fooDirectory2 = directory1.createSubDirectory("foo");
-                fooDirectory2.set("testKey2", "testValue2");
-                assert.strictEqual(fooDirectory, fooDirectory2, "Created two separate subdirectories");
-                assert.strictEqual(fooDirectory.get("testKey2"), "testValue2", "Value 2 not present");
-                assert.strictEqual(fooDirectory2.get("testKey"), "testValue", "Value 1 not present");
+                try {
+                    directory1.createSubDirectory("foo")
+                    assert.fail("Directory creation should fail");
+                } catch (err: any) {
+                    assert.strictEqual(err.message, "Directory with name foo already exists", "Creation should fail");
```

Comment:
```suggestion
                    assert.strictEqual(err.message, "Directory with name foo already exists", "Unexpected error thrown");
```
Nit: seems like a more informative error message for us if the condition being asserted fails.